.github/workflows/deploy-branch.yml

```diff
@@ -1,62 +1,41 @@
-name: "Branch Deployment"
+name: 🌿 Branch Deployment
 
 on:
   workflow_dispatch:
 
 concurrency:
   group: ${{ github.ref }}
 
+permissions:
+  contents: read
+  packages: write
+  id-token: write
+
 jobs:
-  set-deployment-environment:
-    name: Set Deployment Environment
-    runs-on: ubuntu-22.04
-    environment: dev
+  setup-environment:
+    name: 🌱 Set Deployment Environment
+    runs-on: ${{ vars.RUNNER_OS }}
+    environment: ${{ vars.ENVIRONMENT_GITHUB_DEV }}
     steps:
       - name: Set Deployment Environment
         id: set_deployment_env
         run: echo "DEPLOYMENT_ENVIRONMENT=$(echo "${{ github.ref }}" | sed -E 's#refs/heads/(.*)#\1#' | awk '{print tolower($0)}')" >> $GITHUB_OUTPUT
-      - name: Set Terraform Environment Variables
-        id: set_terraform_env_vars
-        run: echo "BLOODCONNECT_DOMAIN=${{ vars.BLOODCONNECT_DOMAIN }}" >> $GITHUB_OUTPUT
     outputs:
       deployment_environment: ${{ steps.set_deployment_env.outputs.deployment_environment }}
-      bloodconnect_domain: ${{ steps.set_terraform_env_vars.outputs.bloodconnect_domain }}
-
-  set-runner-image-name:
-    name: Set Runner Image Name
-    runs-on: ubuntu-22.04
-    steps:
-      - name: Checkout Branch
-        uses: actions/checkout@v3
-
-      - name: Set Runner Image
-        id: set_runner_image
-        run: echo "runner_image_name=ghcr.io/craftsmenltd/bloodconnect-ci-image:$(md5sum ./Dockerfile | awk '{ print $1 }')" >> $GITHUB_OUTPUT
-    outputs:
-      runner_image_name: ${{ steps.set_runner_image.outputs.runner_image_name }}
 
   handle-docker-image:
-    name: Handle Docker Image
-    needs: set-runner-image-name
+    name: 🐳 Handle Docker Image
```

Comment:
same as above last comment

Reply:
updated in f7270e29dddbc6b96c720aaca7302044b7b08e54